### ISSUE TITLE
Plugins in sub-directories ignoring parent plugins as dependencies

### ISSUE THREAD

**Issue by cpeddecord** [fastify/fastify-autoload#137]
## 🐛 Bug Report

Plugins in sub-directories are not respecting dependency chains.

## To Reproduce

Given the following code/directory tree `myPluginFoo` throws an error stating that `myPluginBar` has not been registered:

```typescript
await server.register(autoLoad, {
  dir: path.join(__dirname, 'plugins'),
  maxDepth: 1,
  ignorePattern: /__tests__/,
});
```

```
├── plugins
│   ├── myPluginFoo // has a dep on myPluginBar
│   │   └── index.js
|   └── myPluginBar.js
```

Given the following code with a bit of a shim, dependencies work as expected (note we're disabling `maxDepth`):

```typescript
await server.register(autoLoad, {
  dir: path.join(__dirname, 'plugins'),
  maxDepth: 0,
  ignorePattern: /__tests__/,
});
```

```typescript
// myFooShim.js
export { default } from './myPluginFoo
```

```
├── plugins
│   ├── myPluginFoo // has a dep on myPluginBar
│   |   └── index.js
|   └── myFooShim.js
|   └── myPluginBar.js
```

## Expected behavior

This regression was highlighted during an upgrade from `fastify-autoload` 3.3 -> 3.5.0

## Your Environment

- *node version*: 12
- *fastify version*: >=2.0.0
- *fastify-autoload version*: 3.5.0

**Comment by mcollina** [comment]
Could you amend the ASCII diagrams to clarify in which directory myPluginFoo.js and myPluginBar.js are located? It looks like they are in two at the same time.

A repository to reproduce the problem would be nice.

(Also a PR that fixes it).

Could you check v3.5.1 by any chance?

**Comment by cpeddecord** [comment]
@mcollina updated the ASCII.

Repo here: https://github.com/cpeddecord/fastify-autoload-bug

working through the code at the moment, PR to come. I'll also give the latest code a whirl and see if that doesn't fix it.

**Comment by cpeddecord** [comment]
@mcollina looks like this is addressed by #136, tested `master` branch in my demo repo and it appears to work fine, closing.
